src/main/java/duke/utils/Parser.java

```diff
@@ -0,0 +1,222 @@
+            } catch (DukeException e) {
+                return new InvalidCommand(e.getMessage());
+            }
+        default:
+            return new InvalidCommand("There's no such event type as " + taskType + "!");
+        }
+    }
+
+    /**
+     * Prepares and returns a new DeleteCommand.
+     *
+     * @param userInput Raw user input
+     * @return DeleteCommand based on raw user input
+     */
+    private Command prepareDelete(String userInput) {
+        String[] tokens = userInput.split(" ", 2);
+        if (tokens.length < 2) {
+            return new InvalidCommand("Oh no! Try delete <task index>!");
+        }
+        try {
+            int taskIndex = Integer.parseInt(tokens[1], 10) - 1;
+            return new DeleteCommand(taskIndex);
+        } catch (NumberFormatException e) {
+            return NUMBER_FORMAT;
+        }
+    }
+
+    /**
+     * Prepares and returns a new MarkCommand.
+     *
+     * @param userInput Raw user input
+     * @return MarkCommand based on raw user input
+     */
+    private Command prepareMark(String userInput) {
+        String[] tokens = userInput.split(" ", 2);
+        if (tokens.length < 2) {
+            return new InvalidCommand("Oh no! Try delete <mark index>!");
+        }
+        try {
+            int taskIndex = Integer.parseInt(tokens[1], 10) - 1;
+            return new MarkCommand(taskIndex);
+        } catch (NumberFormatException e) {
+            return NUMBER_FORMAT;
+        }
+    }
+
+    /**
+     * Prepares and returns a new UnmarkCommand.
+     *
+     * @param userInput Raw user input
+     * @return UnmarkCommand based on raw user input
+     */
+    private Command prepareUnmark(String userInput) {
+        String[] tokens = userInput.split(" ", 2);
+        if (tokens.length < 2) {
+            return new InvalidCommand("Oh no! Try delete <unmark index>!");
+        }
+        try {
+            int taskIndex = Integer.parseInt(tokens[1], 10) - 1;
+            return new UnmarkCommand(taskIndex);
+        } catch (NumberFormatException e) {
+            return NUMBER_FORMAT;
+        }
+    }
+
+    private Command prepareFind(String userInput) {
+        String[] tokens = userInput.split(" ", 2);
+        if (tokens.length < 2) {
+            return new InvalidCommand("Oh no! Try find <keyword>!");
+        }
+        return new FindCommand(tokens[1]);
+    }
+
+    /**
+     * Creates and returns LocalDate objects by parsing user's input for date.
+     * User input date should follow the format (yyyy-mm-dd HH:MM), in 24-hour time format.
+     * The date can have a delimiter of a character from "-/|" (characters within the double quotes).
+     * The time can have a delimiter of a character from ":.-|" (characters within the double quotes).
+     * The date and time must be separated by a single whitespace.
+     *
+     * @param dateText User's input of date and time following the format specified.
+     * @return LocalDateTime
+     */
```

Comment:
I like how you have detailed header comments for all your public classes and methods